ted_sws/domain/model/transform.py

```diff
@@ -7,24 +7,64 @@
 
 """ """
 import abc
+from typing import List
 
+from ted_sws.domain.model import PropertyBaseModel
 
-class RuleSet(abc.ABC):
+
+class MappingSuiteComponent(PropertyBaseModel, abc.ABC):
+    class Config:
+        validate_assignment = True
+        orm_mode = True
+
+
+class FileResource(MappingSuiteComponent):
     """
-        A set of rules used in a normalisation or transformation operation.
+
     """
-    version: str = "0.0.1"
+    file_name: str
+    file_content: str
+
+class MetadataConstraints(MappingSuiteComponent):
+    """
+
+    """
+    constraints: dict
 
 
-class TransformationRuleSet(RuleSet):
+class TransformationRuleSet(MappingSuiteComponent):
     """
 
     """
+    resources: List[FileResource]
+    rml_mapping_rules: List[FileResource]
 
 
-class NormalisationRuleSet(RuleSet):
+class SHACLTestSuite(MappingSuiteComponent):
     """
 
     """
+    shacl_tests: List[FileResource]
 
 
+class SPARQLTestSuite(MappingSuiteComponent):
+    """
+
+    """
+    # TODO: Revise this field
+    # purpose: Optional[FileResource] = None
+    sparql_tests: List[FileResource]
+
+
+class MappingSuite(MappingSuiteComponent):
+    """
+
+    """
+    created_at: str
+    identifier: str = "no_id"
```

Comment:
this can be a default optional field: 
`identifier: Optional[str] = uuid.uuid4()`
OR a default mandatory field:
`identifier: str = Field(default=uuid.uuid4())`